kernel/kernel-api/src/main/java/io/delta/kernel/Scan.java

```diff
@@ -56,7 +56,9 @@ public interface Scan {
      *   <li>name: {@code add}, type: {@code struct}</li>
      *   <li>Description: Represents `AddFile` DeltaLog action</li>
      *   <li><ul>
-     *    <li>name: {@code path}, type: {@code string}, description: location of the file.</li>
+     *    <li>name: {@code path}, type: {@code string}, description: location of the file.
+     *    The path is a URI as specified by RFC 2396 URI Generic Syntax, which needs to be decoded
```

Comment:
should the input to Table.forPath also be a String that represents a URI? Have we updated that documentation?

Reply:
It is not at the moment, and we can't change that. If we want to take URI as an input we should be explicit about and and it should be another API, something like `Table.forURI(URI tableURI)`. 

The `path` here comes from Delta Log and stored as a URI in Delta Log according to the protocol. The name is `path` but it is actually a URI. Just updating the documentation to reflect that.

Before the next release, I will have a design decision to change the `path string` to `URI` everywhere else (basically in the `Engine` interfaces). Until then this is the fix.

Reply:
Thanks for explaining!